src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -0,0 +1,139 @@
+// @flow strict
+
+import React, { type Element } from 'react';
+import classNames from 'classnames';
+import { injectIntl, type IntlShape } from 'react-intl';
+import MultiGrid from 'react-virtualized/dist/es/MultiGrid/MultiGrid';
+import AutoSizer from 'react-virtualized/dist/es/AutoSizer';
+import FileIcon from '../../icons/file-icon';
+import messages from '../../elements/common/messages';
+import { getFileExtension } from '../../utils/file';
+import './MetadataBasedItemList.scss';
+
+import type { FlattenedMetadataQueryResponseCollection } from '../../common/types/metadataQueries';
+
+const FILE_ICON_SIZE = 32;
+const FILE_ICON_COLUMN_WIDTH = 52;
+const FILENAME_COLUMN_WIDTH = 350;
+const MIN_METADATA_COLUMN_WIDTH = 250;
+const FIXED_COLUMNS_NUMBER = 2; // 2 Sticky columns - 1. file-icon, 2. file-name for each row
+const FIXED_ROW_NUMBER = 1; // Header row
+
+type Props = {
+    currentCollection: FlattenedMetadataQueryResponseCollection,
+    intl: IntlShape,
+    metadataColumnsToShow: Array<string>,
+} & InjectIntlProvidedProps;
+
+type CellRendererArgs = {
+    columnIndex: number,
+    key: string,
+    rowIndex: number,
+    style: StringAnyMap,
+};
+
+type ColumnWidthCallback = ({ index: number }) => number;
+
+class MetadataBasedItemList extends React.Component<Props> {
+    props: Props;
+
+    getColumnWidth(width: number): ColumnWidthCallback {
+        const { metadataColumnsToShow }: Props = this.props;
+
+        return ({ index }: { index: number }): number => {
+            if (index === 0) {
+                return FILE_ICON_COLUMN_WIDTH;
+            }
+
+            if (index === 1) {
+                return FILENAME_COLUMN_WIDTH;
+            }
+
+            const availableWidth = width - FILENAME_COLUMN_WIDTH - FILE_ICON_COLUMN_WIDTH; // total width minus width of sticky columns
+            // Maintain min column width, else occupy the rest of the space equally
+            return Math.max(availableWidth / metadataColumnsToShow.length, MIN_METADATA_COLUMN_WIDTH);
+        };
+    }
+
+    getGridCellData(columnIndex: number, rowIndex: number): Element<typeof FileIcon> | string {
+        const { currentCollection, metadataColumnsToShow }: Props = this.props;
+        const { items } = currentCollection;
```

Comment:
It looks like we're either checking the length of `items` or defaulting it to an empty array in some places, but not in others.